pvlib/spectrum/mismatch.py

```diff
@@ -689,6 +690,106 @@ def spectral_factor_pvspec(airmass_absolute, clearsky_index,
+        effective irradiance, i.e., the irradiance that is converted to
+        electrical current.
+
+    Notes
+    -----
+    The JRC model parameterises the spectral mismatch factor as a function
+    of absolute air mass and the clear sky index as follows:
+
+    .. math::
+
+        M = 1 + a_1(e^{-k_c}) + a_2(k_c-1)+a_3(AM-1.5),
+
+    where :math:`M` is the spectral mismatch factor, :math:`k_c` is the clear
+    sky index, :math:`AM` is the air mass, :math:`e` is Euler's number, and
+    :math:`a_1, a_2, a_3` are module-specific coefficients. In the JRC model
+    publication, the model used to estimate the air mass (denoted as :math:`AM`
+                                                          ) is not stated. The
+    clear sky index, which is the ratio of GHI to clear sky GHI, uses the ESRA
+    model [2]_ to estimate the clear sky GHI. Prior to this calculation, the
+    irradiance measurements are corrected for angle of incidence using the
+    Martin and Ruiz model [3]_.
+
+    References
+    ----------
+    .. [1] Huld, Thomas, Tony Sample, and Ewan D. Dunlop., 2009. A simple model
+        for estimating the influence of spectrum variations on PV performance.
+        In Proceedings of the 24th European Photovoltaic Solar Energy
+        Conference, Hamburg, Germany pp. 3385-3389. 2009.
+       :doi:`10.4229/24thEUPVSEC2009-4AV.3.27`
+    .. [2] Rigollier, C., Bauer, O. and Wald, L., 2000. On the clear sky model
+       of the ESRA—European Solar Radiation Atlas—with respect to the Heliosat
+       method. Solar energy, 68(1), pp.33-48.
+       :doi:`10.1016/S0038-092X(99)00055-9`
+    .. [3] Martin, N., & Ruiz, J. M., 2001. Calculation of the PV modules
+       angular losses under field conditions by means of an analytical model.
+       Solar Energy Materials and Solar Cells, 70(1), 25-38.
+       :doi:`10.1016/S0927-0248(00)00408-6`
+    """
+
+    _coefficients = {}
+    _coefficients['multisi'] = (0.00172, 0.000508, 0.00000357)
+    _coefficients['cdte'] = (0.000643, 0.000130, 0.0000108)
+
+    if module_type is not None and coefficients is None:
+        coefficients = _coefficients[module_type.lower()]
+    elif module_type is None and coefficients is not None:
+        pass
+    elif module_type is None and coefficients is None:
+        raise ValueError('No valid input provided, both module_type and ' +
+                         'coefficients are None. module_type can be one of ' +
+                         ", ".join(_coefficients.keys()))
+    else:
+        raise ValueError('Cannot resolve input, must supply only one of ' +
+                         'module_type and coefficients. module_type can be ' +
+                         'one of' ", ".join(_coefficients.keys()))
+
+    coeff = coefficients
+    am = airmass
+    kc = clearsky_index
+    mismatch = 1 + coeff[0]*(np.power(math.e, -kc) - np.power(math.e, -1)) +\
+        coeff[1]*(kc-1)+coeff[2]*(am-1.5)
```

Comment:
```suggestion
mismatch = (
        1
        + coeff[0] * (np.power(np.e, -clearsky_index) - np.power(np.e, -1))
        + coeff[1] * (clearsky_index - 1)
        + coeff[2] * (airmass - 1.5)
    )
```
I prefer this way of redacting the equations, I feel it's nicer to read and it doesn't create new variables for brevity. And uses `np.e` instead `math.e` for consistency since numpy already is in the namespace. The value is the same. (feel free to do changes, this is a suggestion only)